### PR TITLE
Download files Modal

### PR DESCRIPTION
**Pull request by christinem**
This PR is based off of #2589 (as it needed the file selector menu changes from that PR). This adds the download file functionality back to the results view, in the form of a modal that can be accessed through a "Download" link in the top right corner.

### CODE REVIEW

File: app/assets/javascripts/Results/index.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+jQuery(document).ready(function () {
+    window.modal_download_files   = new ModalMarkus('#download_files_dialog');
```
Comment (david-yz-liu) [inline app/assets/javascripts/Results/index.js]: remove the unnecessary whitespace

Comment (david-yz-liu) [inline app/assets/javascripts/Results/index.js]: @christinem this is my only comment.

**Comment by david-yz-liu** [comment]
@christinem this is an interesting problem - if the dropdown list is too long, the bottom is inaccessible. (Can't even scroll)

Here is a simple workaround for now: adjust the CSS of this (and only this) modal so that appears at the top, rather than the middle, of the screen.

**Comment by christinem** [comment]
@david-yz-liu Interesting, I never had enough files to notice that. Made that fix, along with fixing merge conflicts and re-arranging the elements on the modal.

**Comment by david-yz-liu** [comment]
Thanks, @christinem.

I suspect there is quite a bit of duplicated code in the different results views, so that's something we should look into at some point.

**Comment by christinem** [comment]
@david-yz-liu Thanks! Yeah, I can see a lot of places where we can put things into partials, definitely would be good to look at.
